File: Sources/YTools/UI/DesignTokens.swift
import SwiftUI

enum DesignTokens {
    static let defaultPanelWidth: CGFloat = 720
    static let panelMaximumHeight: CGFloat = 500
    static let panelMinimumHeight: CGFloat = 196
    static let headerHeight: CGFloat = 72
    static let footerHeight: CGFloat = 34
    static let emptyBodyHeight: CGFloat = 150
    // Includes the plain List's inter-row pitch. The visible backgrounds are
    // 52pt / 44pt; SwiftUI contributes the remaining 8pt between rows.
    static let comfortableRowHeight: CGFloat = 60
    static let compactRowHeight: CGFloat = 52
    static let panelCornerRadius: CGFloat = 14
    static let resultCornerRadius: CGFloat = 9
    static let resultIconSize: CGFloat = 34
    static let horizontalPadding: CGFloat = 18
}

extension AppAccentColor {
    var color: Color {
        switch self {
        case .blue: .blue
        case .purple: .purple
        case .green: .green
        case .orange: .orange
        }
    }
}

extension LauncherAppearanceStyle {
    var headerHeight: CGFloat {
        switch self {
        case .minimal: 40
        case .classic: 60
        case .modern: 68
        case .glass: 64
        }
    }

    var searchFontSize: CGFloat {
        switch self {
        case .minimal: 18
        case .classic: 22
        case .modern, .glass: 23
        }
    }

    var showsFooter: Bool { self == .modern }
    var collapsesWhenIdle: Bool { self != .modern }

    var emptyBodyHeight: CGFloat {
        switch self {
        case .minimal, .glass: 78
        case .classic: 96
        case .modern: DesignTokens.emptyBodyHeight
        }
    }

    var selectionOpacity: Double {
        switch self {
        case .minimal: 0.12
        case .classic: 0.18
        case .modern: 0.16
        case .glass: 0.20
        }
    }

    var backgroundStyle: AnyShapeStyle {
        switch self {
        case .minimal, .modern:
            AnyShapeStyle(.regularMaterial)
        case .classic:
            AnyShapeStyle(.thickMaterial)
        case .glass:
            AnyShapeStyle(.ultraThinMaterial)
        }
    }
}

File: src/YTools.Windows/UI/DesignTokens.cs
using YTools.Models;

namespace YTools.UI;

public static class DesignTokens
{
    public const double DefaultPanelWidth = 720;
    public const double PanelMaximumHeight = 500;
    public const double PanelMinimumHeight = 196;
    public const double HeaderHeight = 72;
    public const double FooterHeight = 34;
    public const double EmptyBodyHeight = 150;
    public const double ComfortableRowHeight = 60;
    public const double CompactRowHeight = 52;
    public const double ResultCornerRadius = 9;
    public const double ResultIconSize = 34;
    public const double HorizontalPadding = 18;

    public static double HeaderHeightFor(LauncherAppearanceStyle style)
    {
        return style switch
        {
            LauncherAppearanceStyle.Minimal => 40,
            LauncherAppearanceStyle.Classic => 60,
            LauncherAppearanceStyle.Modern => 68,
            LauncherAppearanceStyle.Glass => 64,
            _ => 64
        };
    }

    public static double SearchFontSizeFor(LauncherAppearanceStyle style)
    {
        return style switch
        {
            LauncherAppearanceStyle.Minimal => 18,
            LauncherAppearanceStyle.Classic => 22,
            _ => 23
        };
    }

    public static bool ShowsFooter(LauncherAppearanceStyle style)
    {
        return style == LauncherAppearanceStyle.Modern;
    }

    public static bool CollapsesWhenIdle(LauncherAppearanceStyle style)
    {
        return style != LauncherAppearanceStyle.Modern;
    }

    public static double EmptyBodyHeightFor(LauncherAppearanceStyle style)
    {
        return style switch
        {
            LauncherAppearanceStyle.Minimal or LauncherAppearanceStyle.Glass => 78,
            LauncherAppearanceStyle.Classic => 96,
            _ => EmptyBodyHeight
        };
    }

    public static double SelectionOpacity(LauncherAppearanceStyle style)
    {
        return style switch
        {
            LauncherAppearanceStyle.Minimal => 0.12,
            LauncherAppearanceStyle.Classic => 0.18,
            LauncherAppearanceStyle.Modern => 0.16,
            LauncherAppearanceStyle.Glass => 0.20,
            _ => 0.16
        };
    }
}
